src/browser/ui/ReactMount.js

```diff
@@ -451,12 +458,27 @@ var ReactMount = {
    * perform an update on it and only mutate the DOM as necessary to reflect the
    * latest React component.
    *
+   * @param {ReactComponent} parentComponent The conceptual parent of this render tree.
    * @param {ReactElement} nextElement Component element to render.
    * @param {DOMElement} container DOM element to render into.
    * @param {?function} callback function triggered on completion
    * @return {ReactComponent} Component instance rendered in `container`.
    */
-  render: function(nextElement, container, callback) {
+  renderSubtreeIntoContainer: function(parentComponent, nextElement, container, callback) {
+    console.log(typeof parentComponent._reactInternalInstance);
+    invariant(
+      parentComponent != null && parentComponent._reactInternalInstance != null,
+      'parentComponent must be a valid React Component'
+    );
+    invariant(
+      ReactContext.inContextLifecycle,
+      'Must be in a lifecycle method to invoke renderSubtreeIntoContainer'
```

Comment:
What happens if I call this twice in row? Or in the callback? is the `inContextLifecyle` flag going to be correct? My gut tells me that a nested render cycle is going to screw with that global state.

Further, what "lifecycle method" is this safe in? `componentWillReceiveProps`?